generator/transit_generator.cpp

```diff
@@ -43,20 +59,82 @@ string GetFileName(string const & filePath)
   return name;
 }
 
-/// \brief Reads from |root| (json) and serializes an array to |serializer|.
-/// \param handler is function which fixes up vector of |Item|(s) after deserialization from json
-/// but before serialization to mwm.
 template <class Item>
-void SerializeObject(my::Json const & root, string const & key, Serializer<FileWriter> & serializer,
-                     function<void(vector<Item> &)> handler = nullptr)
+void DeserializeFromJson(my::Json const & root, string const & key, vector<Item> & items)
 {
-  vector<Item> items;
-
+  items.clear();
   DeserializerFromJson deserializer(root.get());
   deserializer(items, key.c_str());
+}
+
+void DeserializerGatesFromJson(my::Json const & root, string const & mwmDir, string const & countryId,
+                               vector<Gate> & gates)
+{
+  DeserializeFromJson(root, "gates", gates);
+
+  // Creating IndexRouter.
+  CountryFile const countryFile(countryId);
+  LocalCountryFile localCountryFile(mwmDir, countryFile, 0 /* version */);
+  localCountryFile.SyncWithDisk();
+  CHECK_EQUAL(localCountryFile.GetFiles(), MapOptions::MapWithCarRouting,
+              ("No correct mwm corresponding local country file:", localCountryFile,
+               ". Path:", my::JoinFoldersToPath(mwmDir, countryId + DATA_FILE_EXTENSION)));
+
+  Index index;
+  pair<MwmSet::MwmId, MwmSet::RegResult> const result = index.Register(localCountryFile);
```

Comment:
I see this pattern (create Index, register single mwm, use it) multiple times in generator. Please, abstract it out.

Reply:
Yes. I found three cases in generator: routing index, cross mwm and here (transit). Let's do that. I can do something like that:
```c++
Index index;
RegisterSingleMwmAndCheckIfItWasSuccess(index, ....);
// Using index...
```
I need |index| create at this scope, because then |index| is passed to IndexRouter by ref and kept in IndexRouter by ref. So IndexRouter is valid until |index| is valid.

Reply:
Done.